pkg/sink/heroku.go

```diff
@@ -0,0 +1,44 @@
+package sink
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/davecgh/go-spew/spew"
+	heroku "github.com/heroku/heroku-go/v5"
+	"github.com/pkg/errors"
+)
+
+type HerokuSink struct {
+	BaseSink    `yaml:",inline"`
+	Client      *heroku.Service `yaml:"client"`
+	AppIdentity string          `yaml:"AppIdentity"`
+}
+
+func (sink *HerokuSink) WithHerokuClient(client *heroku.Service) *HerokuSink {
+	sink.Client = client
+	return sink
+}
+
+// Write writes the value of the env var with the specified name for the given repo
+func (sink *HerokuSink) Write(ctx context.Context, name string, val string) error {
+	keypair := map[string]*string{
+		name: &val,
+	}
+	if sink.Client == nil {
+		return errors.New("Heroku Client not set")
+	}
+	updateResult, err := sink.Client.ConfigVarUpdate(ctx, sink.AppIdentity, keypair)
+	if err != nil {
+		return errors.Wrapf(err, "Unable to update Config var with %s:%s", name, val)
+	}
+	fmt.Println("updateResult:")
+	spew.Dump(updateResult)
```

Comment:
curious, what is in this updateResult var?

Reply:
There's not much documentation about this! It's a generic string to string map. This is what I saw from the heroku package:
```

type ConfigVarUpdateResult map[string]*string

// Update config-vars for app. You can update existing config-vars by
// setting them again, and remove by setting it to `null`.
func (s *Service) ConfigVarUpdate(ctx context.Context, appIdentity string, o map[string]*string) (ConfigVarUpdateResult, error) {
	var configVar ConfigVarUpdateResult
	return configVar, s.Patch(ctx, &configVar, fmt.Sprintf("/apps/%v/config-vars", appIdentity), o)
}

```

Reply:
Spew dump I got:
```
updateResult:
(v5.ConfigVarUpdateResult) (len=2) {
 (string) (len=2) "id": (*string)(0xc00045a8d0)((len=15) "missing_version"),
 (string) (len=5) "error": (*string)(0xc00045a8f0)((len=123) "Please specify a version along with Heroku's API MIME type. For example, `Accept: application/vnd.heroku+json; version=3`.\n")
}
```